src/core/lib/security/credentials/external/external_account_credentials.cc

```diff
@@ -430,8 +456,16 @@ void ExternalAccountCredentials::ImpersenateServiceAccount() {
   headers[1].key = gpr_strdup("Authorization");
   headers[1].value = gpr_strdup(str.c_str());
   request.hdrs = headers;
+  std::vector<std::string> body_members;
   std::string scope = absl::StrJoin(scopes_, " ");
-  std::string body = absl::StrFormat("scope=%s", scope);
+  body_members.push_back(absl::StrFormat("scope=%s", UrlEncode(scope).c_str()));
```

Comment:
why was the urlEncode added here?

Reply:
Added UrlEncode as the content-type for the request is x-www-url-formencoded.